### DISCUSSION TITLE
Problem with background map (after update released)

### DISCUSSION THREAD

**Discussion by AfxentiouNikolas** [iTwin/itwinjs-core#3529] [Q&A] [answered]
Hii, before the update released my project was running normally. Now, with the update 3.0 when I start my project and I get this error. Can anyone help me with this error?  (no background map appears)
Thank you in advanced!
![Capture](https://user-images.githubusercontent.com/97607951/163565507-f69cd80c-d1b3-4ac4-9de2-06b0cb410b19.PNG)

**Comment by pmconne** [comment, marked as answer]
As of 3.0, you must [supply your own API keys](https://www.itwinjs.org/changehistory/3.0.0/#removed-default-api-keys) for background maps and terrain.

**Reply by AfxentiouNikolas** [reply]
> As of 3.0, you must [supply your own API keys](https://www.itwinjs.org/changehistory/3.0.0/#removed-default-api-keys) for background maps and terrain.

Where can I find a similar example so I can know where to  find and add these API keys ?
Thank you in advanced!

**Reply by pmconne** [reply]
I literally provided you a link to an example of how to add these API keys. If that doesn't help, provide information about your app, e.g., is it using iTwin Viewer?
You can obtain Bing Maps keys from [Microsoft](https://www.microsoft.com/en-us/maps/create-a-bing-maps-key#:~:text=Sign%20in%20to%20the%20Bing%20Maps%20Dev%20Center%20with%20your%20Microsoft%20Account.&text=Go%20to%20%E2%80%9CMy%20Account%E2%80%9D%20and,click%20on%20%E2%80%9CMy%20Keys.%E2%80%9D&text=Fill%20out%20the%20form%20and,get%20your%20API%20key%20details.).

**Reply by AfxentiouNikolas** [reply]
Thank you so much for you time!
